luaui/configs/buildmenu_sorting.lua

```diff
@@ -769,14 +770,14 @@ local unitOrderTable = {
    ['coratl']         = 260600,
 }
 
+---@type table<string, number>
 local newUnitOrder = {}
 for id, value in pairs(unitOrderTable) do
 	if UnitDefNames[id] then
 		newUnitOrder[UnitDefNames[id].id] = value
 	end
 end
 unitOrderTable = newUnitOrder
-newUnitOrder = nil
```

Comment:
where did this go?

Reply:
I removed it because it would mean adding `nil` to the variable type. I don't believe it's doing anything. As far as I can tell the variable goes out of scope once the module is imported, and it's referencing an object that is referenced by the return anyway.

Reply:
I'm no expert in lua internals, even though it looks safe to remove, it could have unintended consequences (re: memory management) and it's out of scope for the PR as well so personally I wouldn't put my hand on the fire for that line (I guess @Ruwetuin may know better if the line is there because of something since I believe he did that code originally).

Reply:
You can see the same pattern used here with no need for assigning nil:

https://github.com/beyond-all-reason/Beyond-All-Reason/blob/55e3c659e614c502a7de1dee03b15f14f9bb96a6/gamedata/icontypes.lua#L3454-L3463

I'm quite sure it's fine.

Reply:
This is indeed odd. Since Lua tables are passed by reference, 
```lua
unitOrderTable = newUnitOrder
newUnitOrder = nil
```
should set `unitOrderTable` to nil.... right?

Reply:
> This is indeed odd. Since Lua tables are passed by reference,
> 
> ```lua
> unitOrderTable = newUnitOrder
> newUnitOrder = nil
> ```
> 
> should set `unitOrderTable` to nil.... right?

No, that is not how references work. The variable itself holds the memory address of memory in the heap, in this case a table. When assign nil, the memory address is reset to zero so it's no longer referencing the table.

```lua
local a = { x="hello" }
local b = a

-- a and b both point to the same table

local c = b
b = {}
a = nil

-- c now points to the original table
-- b now points to a new empty table
-- a not points to nothing

assert(c.x == "hello")
assert(#b == 0)
assert(a == nil)
```

No, there is no reason to set every variable to nil and the end of its use. The line I removed is not necessary. The variable itself ceases to exist at the end of the file, after it is run.

I'm going to resolve this as no action to take, but I'll move the cleanup changes into a different PR.